File: android2_lesson06_2/src/java/java/com/example/ama/android2_lesson06_2/ui/details/adapter/DetailsAdapter.java
package com.example.ama.android2_lesson06_2.ui.details.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.ama.android2_lesson06_2.model.DeviceParam;

import java.util.List;

public class DetailsAdapter extends ArrayAdapter<DeviceParam> {
    private List<DeviceParam> params;

    public DetailsAdapter(@NonNull Context context, List<DeviceParam> params) {
        super(context, android.R.layout.simple_list_item_2, params);
        this.params = params;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext())
                    .inflate(android.R.layout.simple_list_item_2, parent, false);
        }
        ((TextView) convertView.findViewById(android.R.id.text1))
                .setText(params.get(position).getParamName());
        ((TextView) convertView.findViewById(android.R.id.text2))
                .setText(params.get(position).getParamValue());
        return convertView;
    }
}

File: android2_lesson06_2/src/kotlin/java/com/example/ama/android2_lesson06_2/ui/details/adapter/DetailsAdapter.kt
package com.example.ama.android2_lesson06_2.ui.details.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.example.ama.android2_lesson06_2.model.DeviceParam

class DetailsAdapter(context: Context, val param: List<DeviceParam>)
    : ArrayAdapter<DeviceParam>(context, android.R.layout.simple_list_item_2, param) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            view = LayoutInflater.from(context)
                    .inflate(android.R.layout.simple_list_item_2, parent, false)
        }
        view!!.findViewById<TextView>(android.R.id.text1).text = param[position].name
        view.findViewById<TextView>(android.R.id.text2).text = param[position].value
        return view
    }
}